File: Chap6PE/ex5.cpp
#include <iostream>

using namespace std;

enum TestEnum
{
    ZERO,
    ONE,
    TWO,
    THREE,
    FOUR,
    FIVE,
    SIX,
    SEVEN,
    EIGHT,
    NINE
};

int main(int argc, char const *argv[])
{
    cout<< ONE + TWO <<endl; // 3
    cout<< SIX - THREE <<endl; // 3
    cout<< FIVE * EIGHT <<endl // 40
    cout<< SIX / TWO <<endl; // 3
    /*
    Los tipos enumerados soportan las operaciones
    basicas sobre numeros
    */
    TestEnum test = ONE;
    cout<< test<<endl; // 1
    test = TWO;
    cout<< test <<endl; // 2
    test = (TestEnum) 5;
    cout<< test <<endl; // 5

    /*
    Soportan instanciación y asignacion de valores
    definidos en la enumeración
    
    No soportan asignaciones de enteros de manera directa
    test = 5; genera un error
    */

    return 0;
}

File: Chap6PE/ex5.cs
using System.IO;
using System;

class Program
{
    enum TestEnum{
        ZERO,
        ONE,
        TWO,
        THREE,
        FOUR,
        FIVE,
        SIX,
        SEVEN,
        EIGHT,
        NINE
    };

    static void Main()
    {
        //Console.WriteLine(TestEnum.ONE + TestEnum.TWO); ERROR
        Console.WriteLine(TestEnum.SIX - TestEnum.THREE); // 3
        //Console.WriteLine(TestEnum.FIVE * TestEnum.EIGHT); ERROR
        //Console.WriteLine(TestEnum.SIX / TestEnum.TWO); ERROR
        /*
            C# solo soporta la operacion de resta en tipos enumerados
        */
        TestEnum test = TestEnum.ONE;
        Console.WriteLine(test); // ONE
        test = TestEnum.TWO; 
        Console.WriteLine(test); // TWO
        test = (TestEnum) 5;
        Console.WriteLine(test); // FIVE
        /*
            Soportan instanciación y asignacion de valores
            definidos en la enumeración
            
            No soportan asignaciones de enteros de manera directa
            test = 5; genera un error de casteo implicito 
        */

    }
}
